### ISSUE TITLE
Newline handling for Microsoft.Testing.Platform command line options

### ISSUE THREAD

**Issue by bradwilson** [microsoft/testfx#3645]
## Summary

To make the helper text for command line options more readable, I'd like to be able to pass one or more `Environment.NewLine`s in the command line option description and have them formatted with appropriate indentation.

## Background and Motivation

Some of the [descriptions for xUnit.net's command line option descriptions](https://github.com/microsoft/testfx/issues/3478#issuecomment-2297886957) are very long (due to being complex), and would be dramatically helped with some formatting.

## Proposed Feature

When command line option descriptions include new lines, indent those new lines appropriately.

As examples, here's what we show for `-methodDisplayOptions` in our native CLI:

```
  -methodDisplayOptions <option> : alters the default test display name
                                 : note: you can specify more than one flag by joining with commas
                                 :   none                       - apply no alterations [default]
                                 :   all                        - apply all alterations
                                 :   replacePeriodWithComma     - replace periods in names with commas
                                 :   replaceUnderscoreWithSpace - replace underscores in names with spaces
                                 :   useOperatorMonikers        - replace operator names with operators
                                 :                                  'lt' becomes '<'
                                 :                                  'le' becomes '<='
                                 :                                  'eq' becomes '='
                                 :                                  'ne' becomes '!='
                                 :                                  'gt' becomes '>'
                                 :                                  'ge' becomes '>='
                                 :   useEscapeSequences         - replace ASCII and Unicode escape sequences
                                 :                                   X + 2 hex digits (i.e., 'X2C' becomes ',')
                                 :                                   U + 4 hex digits (i.e., 'U0192' becomes 'ƒ')
```

This is what we show today as one single giant line in the M.T.P CLI:

`--method-display-options     Alters the default test display name. The available values are: 'none' (no altertions) [default], 'all' (all alterations), or one of more of: 'replacePeriodWithComma' (replace periods in names with commas), 'replaceUnderscoreWithSpace' (replace underscores in names with spaces), 'useOperatorMonikers' (replace 'lt' with '<', 'le' with '<=', 'eq' with '=', 'ne' with '!=', 'gt' with '>', and 'ge' with '>='), 'useEscapeSequences' (replace ASCII and Unicode escape sequences; i.e., 'X2C' becomes ',', 'U0192' becomes 'ƒ').`

I think it would be very helpful if I could attempt to replicate the native CLI output here, but today new lines are just written raw to the console output without indenting, so it's not possible for me to recreate this experience.

**Comment by nohwnd** [comment]
I would also like to re-format the output so it is not pushed to right by the longest parameter name. 

![image](https://github.com/user-attachments/assets/b7b3f011-c9b7-4445-9a2b-4d467a05ac8a)

**Comment by nohwnd** [comment]
@bradwilson the newlines are working for me, when I add some newlines into resources they are rendered correctly. 

here I am breaking one of our resources to show it:

![image](https://github.com/user-attachments/assets/11cdc137-ff66-4533-844b-5dd4190387a5)


Was this issue about having correct indentation after a new line? 

In that case #3649 will fix this.

**Comment by bradwilson** [comment]
Yes, this is about indentation.

**Comment by nohwnd** [comment]
Fixed in #3649
